### DISCUSSION TITLE
How to disable CSRF Token for all apis routes in adonis 5

### DISCUSSION THREAD

**Discussion by pk4all** [adonisjs/core#1492] [Help]
I want to disable CSRF Token for all Apis. I have used 

```javascript
exceptRoutes: [
    '/api/*'
  ]
```
But it not working for me. How can I do this, Please help?
Or we user all routes in exceptRoutes.

**Comment by pk4all** [comment]
@thetutlage Any help for this? I am using postman for API check .

**Reply by thetutlage** [reply]
The `exceptRoutes` array takes an actual route pattern and not a glob. `/api/*` is a glob pattern. Instead, I can add support for functions and then inside a function you can decide which routes to allow.

```ts
exceptRoute: function (ctx) {
  return ctx.request.url().includes('/api')
}
```

Can you please create this as an issue on `@adonisjs/shield` repo, so that I don't miss it

**Reply by pk4all** [reply]
Thanks for your reply, I have created an issue on `@adonisjs/shield` repo.
